File: algorithm/48. Rotate Image/48.py
class Solution(object):
    def rotate(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: void Do not return anything, modify matrix in-place instead.
        """
        for i in range(0,len(matrix[0])/2):
            for j in range(0, len(matrix[0])):
                matrix[i][j], matrix[len(matrix[0])-1-i][j] = matrix[len(matrix[0])-1-i][j], matrix[i][j]
        for i in range(0,len(matrix[0])):
            for j in range(i+1, len(matrix[0])):
                matrix[i][j] , matrix[j][i] =matrix[j][i],matrix[i][j]

File: algorithm/48. Rotate Image/48.java
class Solution {
    public void rotate(int[][] matrix) {
        for(int i = 0; i < matrix.length/2; i++)
        {
            int[] tem = matrix[i];
            matrix[i] = matrix[matrix.length - 1 - i];
            matrix[matrix.length - 1 - i] = tem;
        }
        for(int i = 0; i < matrix.length; i++)
        {
            for(int j = i + 1; j < matrix.length; j++)
                swap(matrix, i, j);
        }
    }
    
    private void swap(int[][] mat, int a, int b)
    {
        int tem = mat[a][b];
        mat[a][b] = mat[b][a];
        mat[b][a] = tem;
    }
}
